src/lager.erl

```diff
@@ -594,3 +595,34 @@ pr_stacktrace(Stacktrace) ->
 pr_stacktrace(Stacktrace, {Class, Reason}) ->
     lists:flatten(
         pr_stacktrace(Stacktrace) ++ "\n" ++ io_lib:format("~s:~p", [Class, Reason])).
+
+rotate_file(FileName) ->
+    Handlers = lager_config:global_get(handlers),
+    RotateHandlers = lists:filter(
+        fun ({{lager_file_backend, FN}, _, _}) ->
+                FN == FileName orelse 
+                lager_util:expand_path(FN) == lager_util:expand_path(FileName);
+            (_) -> false
+        end,
+        Handlers),
+    rotate_handlers(RotateHandlers).
+
+rotate_sink(Sink) ->
+    Handlers = lager_config:global_get(handlers),
+    RotateHandlers = lists:filter(
+        fun({_,_,S}) -> S == Sink; 
+           (_) -> false 
+        end, 
+        Handlers),
+    rotate_handlers(RotateHandlers).
+
+rotate_all() -> 
+    rotate_handlers(lager_config:global_get(handlers)).
+
+
+rotate_handlers(Handlers) ->
+    [ rotate_handler(Handler) || Handler <- Handlers ].
+
+rotate_handler({Backend, Handler, Sink}) ->
+    gen_event:call(Sink, Backend, rotate, infinity).
```

Comment:
Looks like any 3rd party backends will not know about the new `rotate` message and cause this to lock up indefinitely.

Perhaps it makes more sense to use a `gen_event:cast` for this.